x/pairing/keeper/filters/filter.go

```diff
@@ -0,0 +1,68 @@
+package filters
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	projectstypes "github.com/lavanet/lava/x/projects/types"
+)
+
+type Filter interface {
+	Filter(ctx sdk.Context, providers []epochstoragetypes.StakeEntry) []bool
+	InitFilter(strictestPolicy projectstypes.Policy) bool // return if filter is usable (by the policy)
+}
+
+func GetAllFilters() []Filter {
```

Comment:
why do we need this function, can be merged with initFilters

Reply:
Omer told me to implement FilterProviders with a filters argument (because in the future we might call it with some of the filters).
If I put it inside initFilters, It's going to init unecessary filters